packages/core/src/api/blockManipulation/selections/selection.ts

```diff
@@ -127,8 +143,55 @@ export function getSelection<
     );
   }
 
+  const selectionContent = tr.selection.content().content;
+  selectionContent.descendants((node) => {
+    if (cutStartNode && cutEndNode) {
+      return false;
+    }
+
+    if (
+      node.type.name === startNode.type.name &&
+      node.attrs.id === startNode.attrs.id
+    ) {
+      cutStartNode = node;
+    }
+
+    if (
+      node.type.name === endNode.type.name &&
+      node.attrs.id === endNode.attrs.id
+    ) {
+      cutEndNode = node;
+    }
+
+    return true;
+  });
+
+  const cutBlocks: Block<BSchema, I, S>[] = [];
+  for (let i = 0; i < blocks.length; i++) {
+    if (i === 0) {
+      cutBlocks.push(nodeToBlock(cutStartNode!, pmSchema));
+    } else if (i === blocks.length - 1) {
+      cutBlocks.push(nodeToBlock(cutEndNode!, pmSchema));
+    } else {
+      cutBlocks.push(blocks[i]);
+    }
+  }
+
   return {
     blocks,
+    cutBlocks,
+    blocksCutAtStart:
+      startNode.nodeSize !== cutStartNode!.nodeSize
+        ? $startBlockBeforePos.nodeAfter!.attrs.id
+        : undefined,
+    blocksCutAtEnd:
+      endNode!.nodeSize !== cutEndNode!.nodeSize
+        ? $endBlockBeforePos.nodeAfter!.attrs.id
+        : undefined,
```

Comment:
I'm still going to have to push back on this naming, I don't know what a `blocksCutAtStart` means.

Reply:
OK, I think I understand what cutBlocks are now from the tests. It is the slice of content that the selection range is over. imo, I don't think we should introduce another term on top of what prosemirror has already established for this (a slice).

I think it should be: `blockSlice`, I still don't understand the usefulness of blocksCutAtStart & blocksCutAtEnd, but I think it is something more like `startBlockId` or `sliceBlockId`